querier.go

```diff
@@ -504,6 +504,13 @@ func (s *baseChunkSeries) Next() bool {
 		err      error
 	)
 
+	if lsetLen := len(s.lset); lsetLen > 0 {
```

Comment:
why not just

```
var (
		lset     = make(labels.Labels, len(s.lset))
		chkMetas = make([]chunks.Meta, len(s.chks))
		err      error
	)
```

Reply:
I thought a `make(..., 0)` would allocate memory, since it produces a non-nil slice: https://play.golang.org/p/9ISeSKFbJaM

But [apparently](https://stackoverflow.com/questions/44305170/nil-slices-vs-non-nil-slices-vs-empty-slices-in-go-language) it will simply create a slice with a `Data` pointer referencing some fixed address, so there's no memory penalty in making a zero-sized slice.

Replaced with your suggestion.

Reply:
yeah I also did some benchmarking and it showed the same results.